doc/changes.rst

```diff
@@ -1,12 +1,302 @@
+
+``--sec-severity``
+  * Renamed to ``--advisory-severities``.
+
+``-v, --verbose``
+  * Not implemented at present. May be added for specific commands in the future.
+
+``--version``
+  * Behavior is different now. See the :ref:`main man page <version_option_ref-label>` for more details.
+
+
+Newly introduced options
+^^^^^^^^^^^^^^^^^^^^^^^^
+``--allow-downgrade``
+  * Along with ``--no-allow-downgrade``, these options enable/disable the downgrade of dependencies when resolving transactions.
+  * New respective configuration options have also been created.
+  * Applicable to ``install``, ``upgrade``, and related commands.
+
+``--dump-main-config``
+  * Along with related ``--dump-repo-config=REPO_ID``, these are new options to print configuration values on the standard output.
+
+``--offline``
+  * Store the transaction to be performed offline.
+  * Applicable to all relevant transactional commands.
+  * See the :ref:`Offline command <offline_command_ref-label>` for more information.
+
+``--show-new-leaves``
+  * Show newly installed leaf packages and packages that became leaves after a transaction.
+
+``--skip-unavailable``
+  * Allow skipping packages that are not available in repositories.
+  * Not to be confused with the :ref:`skip_if_unavailable <skip_if_unavailable_options-label>` configuration option.
+  * Applicable to ``install``, ``upgrade``, and related commands.
+  * See also the :ref:`strict option deprecation <strict_option_conf_changes_ref-label>` for more information.
+
+``--use-host-config``
+  * See the :ref:`main man page <use_host_config_option_ref-label>` for more details.
+
+
+Commands
+--------
+
+Optional arguments
+^^^^^^^^^^^^^^^^^^
+Commands cannot have optional subcommands and optional arguments. Optional subcommands were ambiguous,
+making it unclear whether the input was intended as a command argument or a subcommand. Subcommands are now mandatory if present.
+
+Examples:
+  * Before: ``dnf history <transaction ID>`` Now: ``dnf history info <transaction ID>``
+  * Before: ``dnf updateinfo`` Now: ``dnf updateinfo summary``
+
+
+Changes to individual commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``alias``
+  * Dropped. The command has been replaced by a different functionality.
+  * See the :ref:`Aliases for command line arguments <aliases_misc_ref-label>` for more information.
+
+``automatic``
+  * Now a DNF5 plugin.
+  * Different configuration file locations and some changes in format.
+  * See the :ref:`Automatic command <automatic_plugin_ref-label>` for more information.
+
+``autoremove``
+  * Dropped the ``<spec>`` positional argument since the use case is sufficiently covered by the ``remove`` command.
+  * Specific variants of the command, ``autoremove-n``, ``autoremove-na``, and ``autoremove-nevra``, are not supported anymore.
+
+``builddep``
+  * Dropped ``--spec`` and ``--srpm`` arguments as automatic detection from file extensions is implemented now.
```

Comment:
This is probably a bad idea because there is no guarantee we can detect from file extensions. I've seen spec files with all kinds of crazy file names, and while it's not super-common anymore to see `.spm` or similar weird things for SRPMs, it's still possible since RPM allows reconfiguring the package filename template.

Reply:
Hmm, interesting, I wouldn't think of that. This implementation of the `builddep` plugin is there from the beginning and I don't see any explicit request to do it in the automatic way, so we can definitely think about bringing back the original behavior. Could you create an issue for that please?

Reply:
There already is https://github.com/rpm-software-management/dnf5/issues/799 and the associated PR https://github.com/rpm-software-management/dnf5/pull/810